server/src/main/java/org/opensearch/snapshots/SnapshotShardsService.java

```diff
@@ -423,43 +434,72 @@ private void snapshot(
                         commitGeneration = snapshotIndexCommit.getGeneration();
                         indexShard.acquireLockOnCommitData(snapshot.getSnapshotId().getUUID(), primaryTerm, commitGeneration);
                     }
-                    try {
-                        repository.snapshotRemoteStoreIndexShard(
-                            indexShard.store(),
-                            snapshot.getSnapshotId(),
-                            indexId,
-                            snapshotIndexCommit,
-                            getShardStateId(indexShard, snapshotIndexCommit),
-                            snapshotStatus,
-                            primaryTerm,
-                            startTime,
-                            ActionListener.runBefore(listener, wrappedSnapshot::close)
-                        );
-                    } catch (IndexShardSnapshotFailedException e) {
-                        logger.error(
-                            "Shallow Copy Snapshot Failed for Shard ["
-                                + indexId.getName()
-                                + "]["
-                                + shardId.getId()
-                                + "] for snapshot "
-                                + snapshot.getSnapshotId()
-                                + ", releasing acquired lock from remote store"
-                        );
-                        indexShard.releaseLockOnCommitData(snapshot.getSnapshotId().getUUID(), primaryTerm, commitGeneration);
-                        throw e;
-                    }
-                    long endTime = threadPool.relativeTimeInMillis();
-                    logger.debug(
-                        "Time taken (in milliseconds) to complete shallow copy snapshot, "
-                            + "for index "
-                            + indexId.getName()
-                            + ", shard "
-                            + shardId.getId()
-                            + " and snapshot "
-                            + snapshot.getSnapshotId()
-                            + " is "
-                            + (endTime - startTime)
+                    GatedCloseable<IndexCommit> finalWrappedSnapshot = wrappedSnapshot;
```

Comment:
we should throw `IndexShardSnapshotFailedException` along with `listener.onFailure(e);`  https://github.com/opensearch-project/OpenSearch/blob/829215c4a4660ee65026c40a0a01ebe100b3ee2c/server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java#L2691